### PR TITLE
HPCC-25331 Fix packagemap lookup of cluster information in cloud

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by afishbeck** [comment]
@richardkchapman I'm adding some new roxie cluster info, can you verify those 3 values?

File: esp/smc/SMCLib/TpWrapper.cpp [hunk 1]
```diff
@@ -2077,21 +2077,30 @@ class CContainerWUClusterInfo : public CSimpleInterfaceOf<IConstWUClusterInfo>
...
+            thorProcesses.append(name);
         }
         else if (strieq(type, "roxie"))
         {
             agentQueue.set(getClusterEclAgentQueueName(queue.clear(), name));
             platform = RoxieCluster;
+            if (cfg)
+            {
+                roxieRedundancy = (unsigned) cfg->getPropInt("@replicas", roxieRedundancy);
+                channelsPerNode = (unsigned) cfg->getPropInt("@numChannels", channelsPerNode);
```
Comment (richardkchapman) [inline esp/smc/SMCLib/TpWrapper.cpp:2102]: numChannels is total number of channels in the cluster, not per node

channels per node is always 1 in containerized system. frequently 2 but could be more in bare metal

**Review by richardkchapman — CHANGES_REQUESTED** [review]
channelsPerNode code looks wrong

**Comment by richardkchapman** [comment]
@afishbeck

**Comment by afishbeck** [comment]
@richardkchapman updated channelsPerNode.  These values only affect cloud.

File: helm/hpcc/templates/_helpers.tpl [hunk 2]
```diff
@@ -570,6 +570,8 @@ Generate instance queue names
   type: roxie 
   prefix: {{ .prefix | default "null" }}
   queriesOnly: true
+  replicas: {{ .replicas | default 2 }}
```
Comment (richardkchapman) [inline helm/hpcc/templates/_helpers.tpl:573]: Not convinced that this is right. Apart form anything else the default in roxie.yaml is 1.

I don't know if there's some way to make the default be only specified in one place...

**Review by richardkchapman — CHANGES_REQUESTED** [review]
One comment

File: helm/hpcc/templates/_helpers.tpl (hunk 2 above)
Comment (afishbeck) [reply helm/hpcc/templates/_helpers.tpl:573]: Actually, I don't think this replicas value has anything to do with what I want in the cluster info.  I think the cluster info is more data replica?  And in cloud we leave that up to the cloud environment?  I'll remove this setting from here.  And in the cluster info should probably just return 1.

**Comment by afishbeck** [comment]
@richardkchapman updated.

**Review by richardkchapman — APPROVED** [review]

**Review by wangkx — APPROVED** [review]
Looks fine for me

**Pull request merged**